packages/icon/src/vaadin-icon.js

```diff
@@ -110,7 +110,10 @@ class Icon extends ThemableMixin(ElementMixin(ControllerMixin(SlotStylesMixin(Ic
         viewBox="[[__computeViewBox(size, __viewBox)]]"
         preserveAspectRatio="[[__computePAR(__defaultPAR, __preserveAspectRatio)]]"
         aria-hidden="true"
-      ></svg>
+      >
+        <g id="svg-group"></g>
+        <g id="use-group"></g>
```

Comment:
This would probably simplify the implementation a bit since then you don't need a reference to the node

```suggestion
        <g id="use-group" hidden="[[!__useRef]]">
          <use href="[[__useRef]]" />
        </g>
```